includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1613,34 +1615,19 @@ public function set_payment_method_title_for_order( $order, $payment_method_type
 	 * @return array Array of keyed metadata values.
 	 */
 	protected function get_metadata_from_order( $order, $payment_type ) {
+		$service  = wcpay_get_container()->get( OrderService::class );
+		$metadata = $service->get_payment_metadata( $order->get_id(), $payment_type ) ?? [];
```

Comment:
Nitpick: I don't think `$service->get_payment_metadata()` can return `null`. So using null coalescing operator here is a bit extreme. In case, it can return `null`, let's declare that in phpdoc `@return`. 

```suggestion
		$metadata = $service->get_payment_metadata( $order->get_id(), $payment_type )
```

Reply:
Fixed in https://github.com/Automattic/woocommerce-payments/pull/7333/commits/27859cab56c3ee53e1134ad2ffdbc261c0e4caea. Initially, I did this because I was planning to skip calls in tests, then I realized that it wouldn't be right.